cypress/integration/operator_spec.js

```diff
@@ -31,7 +36,68 @@ describe('Operator creation basic tests', function () {
         cy.contains('Omat palvelutiedot');
     });
 
-    it('should validate and invalidate business id', function () {
+    it('should navigate to add new operator', function () {
+        cy.contains('Omat palvelutiedot');
+        cy.get('#btn-add-new-transport-operator').click();
+        cy.contains('Lisää palveluntuottaja');
+    });
+
+    it('should not be able to save operator with invalid business id', function () {
+        cy.contains('Omat palvelutiedot');
+        cy.get('#btn-add-new-transport-operator').click();
+        cy.get('#input-business-id').type('1234');
+        // Check warning
+        cy.contains(testOp1.businessIdInvalidWarding);
+
+        // Save button should be disabled
+        cy.get('#btn-operator-save').should('be.disabled');
+        cy.visit('/#/');
+        cy.get('#btn-confirm-leave-page').click();
+    });
+
+    it('should add new operator', function () {
+
+        cy.contains('Omat palvelutiedot');
+
+        // Start adding new operator
+        cy.get('#btn-add-new-transport-operator').click();
+
+        // Business id is already added
+        cy.get('#input-business-id').type('1234567-8');
+        // Check warning
+        cy.contains(testOp1.businessIdExistsWarning);
+
+        // Try creating random business-id
+        cy.get('#input-business-id').clear(); // Clear business-id
+        cy.get('#input-business-id').type(randomBusinessid());
+        cy.contains(testOp1.businessIdExistsWarning).should('not.exist');
+
+        // Add name
+        const newOperatorName = randomName('ctest-op-');
+        cy.get('#input-operator-name').type(newOperatorName);
+
+        // Save
+        cy.get('#btn-operator-save').click();
+
+        // Ensure that operator is correctly added - should be selected by default
+        cy.wait(500);
+        cy.contains(newOperatorName);
+
+        // Ensure that other pages have it selected by default
+        cy.visit('/#/pre-notices');
+        cy.contains(newOperatorName);
+        cy.visit('/#/routes');
+        cy.contains(newOperatorName);
+
+        // Delete operator
```

Comment:
This should ensure also that deletion works and doesn't produce an error.

Reply:
Added